bundles/plugins/org.bonitasoft.studio.connectors/src-test/java/org/bonitasoft/studio/connectors/ui/property/section/ConnectorEventFilterTest.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (C) 2015 BonitaSoft S.A.
+ * BonitaSoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2.0 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.studio.connectors.ui.property.section;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.bonitasoft.engine.bpm.connector.ConnectorEvent;
+import org.bonitasoft.studio.model.process.Connector;
+import org.bonitasoft.studio.model.process.ProcessFactory;
+import org.bonitasoft.studio.model.process.Task;
+import org.eclipse.core.runtime.AssertionFailedException;
+import org.eclipse.jface.viewers.Viewer;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ConnectorEventFilterTest {
+	
+	
+	 @Mock
+	 private Viewer viewer;
+	 ConnectorEventFilter onFinishFilter;
+	 
+	 @Before
+	 public void setup() {
+		 onFinishFilter = new ConnectorEventFilter(ConnectorEvent.ON_FINISH.toString());
+	 }
+	
+	 
+	@Test
+	public void should_return_true_when_connectorEvent_isOnFinish(){
+		Connector connector = ProcessFactory.eINSTANCE.createConnector();
+		connector.setEvent(ConnectorEvent.ON_FINISH.toString());
```

Comment:
You may use org.bonitasoft.studio.model.process.builders.ConnectorBuilder.
You can then write a one-liner for your test:
assertTrue(onFinishFilter.select(viewer, null, aConnector().onEvent(ConnectorEvent.ON_FINISH.name()).build()));